src/flexmock/api.py

```diff
@@ -424,11 +447,14 @@ def mock_method(runtime_self: Any, *kargs: Any, **kwargs: Any) -> Any:
                 for expectation in reversed(FlexmockContainer.flexmock_objects.get(self, []))
                 if expectation.name == name
             ]
-            error_msg = _format_args(name, arguments)
+            error_msg = (
+                f"Arguments for call {name} did not match expectations:\n"
+                f"  Received call:\t{_format_args(name, arguments)}"
+            )
             if expectations:
-                for expectation in expectations:
-                    error_msg += "\nDid not match expectation %s" % _format_args(
-                        name, expectation._args
+                for index, expectation in enumerate(expectations, 1):
+                    error_msg += (
+                        f"\n  Expected call[{index}]:\t{_format_args(name, expectation._args)}"
```

Comment:
How about something more elegant like
```suggestion
                error_msg += "\n".join(
                     f"  Expected call[{index}]:\t{_format_args(name, expectation._args)}"
                     for index, expectation in enumerate(expectations, 1)
                )
```
(there is a missing new line like this but there are different ways to deal with is to it's just a global idea)

Reply:
Done! https://github.com/flexmock/flexmock/pull/61/commits/1eee0148851941232107852ceeeb448f0155a277